rust/lance/src/index.rs

```diff
@@ -570,7 +570,7 @@ impl DatasetIndexExt for Dataset {
             .len();
 
         let num_unindexed_fragments = self.fragments().len() - num_indexed_fragments;
-        let num_indexed_rows = num_indexed_rows_per_delta.iter().last().unwrap();
+        let num_indexed_rows: usize = num_indexed_rows_per_delta.iter().cloned().sum();
```

Comment:
@chebbyChefNEQ I saw you wrote `.last()` in #2979. Do you remember why this made sense to you at the time?

Reply:
`num_indexed_rows_per_delta` is a vec of cumulative number of rows indexed I think. so we get a vec like
[100, 150, 200] instead of `[100, 50, 50]`

Reply:
so `.last()` is the total number of indexed rows

Reply:
Was it intentional that it was cumulative? I have been treating it as a bug in this PR

Reply:
I'm not sure. I don't think we ever made a contract around this. I agree that cumulative seems like a bug and we should record the number of indexed rows per delta